fix(components/docs-tools): show `?` indicator only for optional interface properties

Blackbaud-SteveBrush opened the pull request:
[AB#3341644](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3341644)

Blackbaud-SteveBrush committed:
fix(components/docs-tools): show `?` optional indicator only for optional interface properties

Blackbaud-SteveBrush commented on eslint-base.config.js:
```diff
@@ -42,15 +42,5 @@ module.exports = [
       ],
     },
   },
-  {
```
This block is duplicated in this config file for some reason. Removing it so there's only one instance.

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/3424/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/3424/storybooks/toast-storybook/)

Apps:

- [code-examples-playground](https://blackbaud.github.io/skyux-pr-preview/3424/code-examples-playground/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/3424/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/3424/playground/)

Blackbaud-SteveBrush committed:
Merge branch 'main' into fix-docs-tools-interface-property-optional

Blackbaud-SandhyaRajasabeson committed:
feat(components/layout): add inline delete harness (#3420)

[AB#2195478](https://dev.azure.com/blackbaud/Products/_boards/board/t/SKY%20UX%20Program/Stories?System.AssignedTo=%40me&workitem=2195478)

---------

Co-authored-by: Sandhya Adhirvuh <sandhya.rajasabeson@blackbaud.me>

Blackbaud-SteveBrush committed:
Merge remote-tracking branch 'origin/main' into fix-docs-tools-interface-property-optional

johnhwhite committed:
Merge branch 'main' into fix-docs-tools-interface-property-optional
